dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -253,9 +255,18 @@ private String resultProcess(ResultSet resultSet) throws Exception {
                 : JSONUtils.toJsonString(resultJSONArray);
 
         if (Boolean.TRUE.equals(sqlParameters.getSendEmail())) {
+            resultJSONArray.forEach(row -> {
+                row.fields().forEachRemaining(field -> {
+                    if (field.getValue() instanceof NullNode) {
+                        field.setValue(new TextNode("null"));
+                    }
+                });
+            });
+            String sendResult = resultJSONArray.isEmpty() ? JSONUtils.toJsonString(generateEmptyRow(resultSet))
+                    : JSONUtils.toJsonString(resultJSONArray);
```

Comment:
Please add some unit-test for this.

Reply:
I don't know how to write a unit test for this piece.

Reply:
I started all the projects and tested them with simulated data.

Reply:
> I don't know how to write a unit test for this piece.

You can take a look at other UT.

> I started all the projects and tested them with simulated data.

Testing is used to ensure the normal operation in the future, and the current normal operation cannot guarantee that the future will not be affected by other factors.